packages/repository/src/repositories/repository.ts

```diff
@@ -250,16 +251,18 @@ export class CrudRepositoryImpl<T extends Entity, ID>
     return this.toModels(this.connector.find(this.model, filter, options));
   }
 
-  findById(id: ID, options?: Options): Promise<T> {
+  async findById(id: ID, filter?: Filter, options?: Options): Promise<T> {
     if (typeof this.connector.findById === 'function') {
       return this.toModel(this.connector.findById(this.model, id, options));
     }
     const where = this.model.buildWhereForId(id);
-    return this.connector
-      .find(this.model, {where: where}, options)
-      .then((entities: T[]) => {
-        return entities[0];
-      });
+    const entities = await this.toModels(
+      this.connector.find(this.model, {where: where}, options),
+    );
+    if (!entities.length) {
+      throw new EntityNotFoundError(this.model.name, id);
```

Comment:
can we leverage `this.model.modelName()` here?

Reply:
I am going to fix the type of the constructor argument `model` and use `this.model` (let EntityNotFoundError to obtain the model name).